packages/ui/test/working-groups/modals/AddWorkerStakeModal.test.tsx

```diff
@@ -0,0 +1,194 @@
+    showModal: jest.fn(),
+  }
+
+  beforeAll(async () => {
+    await cryptoWaitReady()
+    useMyAccounts.allAccounts.push(alice, bob)
+  })
+
+  beforeEach(async () => {
+    stubDefaultBalances(api)
+    transfer = stubTransaction(api, 'api.tx.storageWorkingGroup.increaseStake')
+  })
+
+  describe('Form step', () => {
+    it('Renders modal', () => {
+      const { queryByText } = renderModal()
+
+      expect(queryByText('Add stake')).toBeDefined()
+    })
+
+    it('Setup minimal stake', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      expect(input?.value).toBe('200')
+      expect(submitButton).not.toBeDisabled()
+    })
+
+    it('Too low stake', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+
+      fireEvent.change(input, { target: { value: '100' } })
+
+      expect(submitButton).toBeDisabled()
+    })
+
+    it('Stake exceed balance', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+
+      fireEvent.change(input, { target: { value: '10000' } })
+
+      expect(submitButton).toBeDisabled()
+    })
+  })
+
+  describe('Sign step', () => {
+    it('Goes to sign modal', async () => {
+      const { queryByText } = await renderModalAndProceedToSign()
+
+      expect(queryByText('Authorize transaction')).toBeDefined()
+    })
+
+    it('Transaction exceed balance', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+
+      fireEvent.change(input, { target: { value: '1000' } })
+      submitButton.click()
+
+      const signButton = await getButton('Sign transaction and Stake')
+
+      expect(signButton).toBeDisabled()
+    })
+
+    describe('Pending', () => {
+      it('Renders wait for transaction', async () => {
+        stubTransactionPending(transfer)
+
+        await renderModalAndProceedToSign(true)
+
+        expect(screen.queryByText('Pending transaction')).toBeDefined()
+      })
+    })
+
+    describe('Success', () => {
+      beforeEach(async () => {
+        stubTransactionSuccess(transfer, 'storageWorkingGroup', 'StakeIncreased')
+      })
+
+      it('Renders transaction success', async () => {
+        await renderModalAndProceedToSign(true)
+
+        expect(await screen.findByText('Success')).toBeDefined()
+      })
+    })
+
+    describe('Failure', () => {
+      it('Renders transaction failure', async () => {
+        stubTransactionFailure(transfer)
+        await renderModalAndProceedToSign(true)
+
+        expect(await screen.queryByText('Failure')).toBeDefined()
+      })
+    })
+  })
+
+  const renderModalAndProceedToSign = async (shouldSign = false) => {
```

Comment:
The usage outside looks like this:
```ts
renderModalAndProceedToSign(true)
// or:
renderModalAndProceedToSign()
```

It is not totally clear what it will do, a more fluent example would be:

```ts
renderModalAndProceedToSign()
signTransaction()
```
OR
```ts
renderModal()
passStakeStep()
signTransaction()
```